### DISCUSSION TITLE
How do I override ‘special’ scaling unit values?

### DISCUSSION THREAD

**Discussion by kokoshneta** [tailwindlabs/tailwindcss#9463] [Help] [answered]
The `theme.spacing` values that can be overridden in your Tailwind config file are [listed in the documentation](https://tailwindcss.com/docs/customizing-spacing#default-spacing-scale), but these only include the ‘general’ spacing units – plain numbers and the value `px`.

In various different properties, however, there are additional ‘special’ spacing units, such as `prose` (used in `max-w-prose`), `full` (used in `w-full`, `h-full`, etc.) and so on, and these seem to override the general units.

How can I add or override units in my config file so that they apply everywhere?

For example, if I want the unit `prose` to be 50ch and be available for properties in general (`w-prose`, `h-prose`, `p-prose`, etc.), I can add it to `theme.extend.spacing`. But `max-w-prose` will still be 65ch, because that’s apparently a hard-coded value for that whole string, rather than being made up of the `max-w-` rule + the general unit `prose`.

Is there a way to add a unit in such a way that it will work with _all_ properties that expect a spacing unit, including those that already have a built-in value for that unit? Preferably one that doesn’t involve manually overriding each of those values individually.

**Comment by wongjn** [comment, marked as answer]
We can look at the default configuration and how it is built to give a clue on how we can achieve your desired configuration:

https://github.com/tailwindlabs/tailwindcss/blob/8de17cb6482714977e0317a20a2c6690c629eae8/stubs/defaultConfig.stub.js#L708

If we compare this to say, the width utilities:

https://github.com/tailwindlabs/tailwindcss/blob/8de17cb6482714977e0317a20a2c6690c629eae8/stubs/defaultConfig.stub.js#L890-L924

We can see the width configuration is resolved via `spacing` but also its own properties. So to only have the `spacing` units like with the padding utilities, we can **override** the default configuration for the desired properties to only have values from spacing, like so:

```js
module.exports = {
  theme: {
    width: ({ theme }) => theme('spacing'), 
    height: ({ theme }) => theme('spacing'), 
    maxWidth: ({ theme }) => theme('spacing'), 
    // And so on for the properties you'd like. 
  },
};
```
Ensure the configuration for these are under `theme` and **not** `theme.extend`, so that these values will **replace** and not **add to**.

**Reply by kokoshneta** [reply]
Ah, that’s definitely better than the workaround I had in mind. It does still require identifying which properties don’t use the spacing units and then adding them in manually, but this should be more doable.

(Note that I _don’t_ want to replace the entire built-in configurations, only add to them and override individual values, so I add it to `theme.extend`. I still want to be able to use `min-w-max`, `max-w-full` and so on.)
